Singularity-3revert

megosato opened the pull request:
…63182e7c65a15e4a102087f to d019c3328d6395e33e35e5f2ddfc7059299238fd (inclusive) (#264)"

This reverts commit a0ebabe70b04c99ae01ba5070d1f4fdaaf4badfe.

robbason commented on ever_given/ever_given/engines/singularity_engine.py:
```diff
@@ -0,0 +1,124 @@
+from .utils import ContainerInstance, Engine, GUEST_OUTPUT_DIR
+
+AFTER_TERMINATE_WAIT = 5  # Seconds to wait after 'terminate' signal before 'kill'
+SINGULARITY_CONTAINER_TYPES = ["docker", "singularity", "singularity_sif"]
```
```suggestion
SINGULARITY_CONTAINER_TYPES = ["docker", "singularity_remote", "singularity_local"]
```

robbason commented on ever_given/ever_given/tests/test_cancellation.py:
```diff
@@ -1,23 +1,41 @@
+        assert os.path.exists(container_uri)
+    if container_engine == "docker":
+        command = f"docker pull {container_uri}"
+        proc = subprocess.Popen(command, shell=True)
```
use subprocess.run here and above

megosato commented:
@robbason addressed your comments from here: https://github.com/samplchallenges/SAMPL-league/pull/278

I was using that branch to compare time if I downloaded all the sif files from ever_given once to see if it significantly improved testing time.

megosato commented on app/sampl/base_settings.py:
```diff
@@ -100,17 +101,50 @@
 
-LOGIN_TO_AWS = False
+# container engine is either "docker" or "singularity"
+CONTAINER_ENGINE = "docker"
```
@robbason should this (CONTAINER_ENGINE) be pulled from environment variables?

robbason replied:
Let's discuss it in the context of the remote scheduler setup. In my opinion, we could go two ways:
1- environment variables all the way
2 - env variables for things that shouldn't be in config files, like hostnames and passwords, and use different django settings modules for settings that should be grouped together. For example, if you have a remote scheduler, container engine should be singularity.
